parcellab/common/templates/_configmap.tpl

```diff
@@ -9,10 +9,13 @@
   ) }}
 */}}
 {{- define "common.configmap" -}}
-{{- if or .Values.config .config }}
+{{- if or .config .Values.config }}
 {{- $componentValues := (merge (deepCopy .) (dict "component" .name)) -}}
 {{- $name := include "common.componentname" $componentValues -}}
-{{- $config := default .Values.config .config -}}
+{{- $config := .config -}}
+{{- if and (not (empty .Values)) (not (empty .Values.config)) -}}
+{{- $config = .Values.config -}}
+{{- end -}}
```

Comment:
Isn't and if/else if better than if or?

Reply:
yes, but the problem is, that the .Values can be null when getting values only through `.config`. An if/else would be better readable for sure, but helm needs to be ensured the $config is initialised. And with that constraint I thought this might be  a readable solution although not beautiful.

Reply:
There is something I am missing with this change.

Originally:

```
{{- $config := default .Values.config .config -}}
```

The intention of that is either having `.config` (pass specifically) or `.Values.config` if that config map is not passed specifically. Basically, .Values.config is the kitchen sink, and if that one is specified, then it is a problem with `values.yaml` because it should always define `.Values.config` in its default values.

Reply:
One of the reasons of enforcing the configuration of some enforced values in `values.yaml` is to not complicate this more, as it is traditionally already complicated: because we are dealing with a template language.

But maybe I am missing another reason

Reply:
The problem with the original implementation is, that the entire `.Values` cannot be null because the check on `.Values.config` would break.  

BUT - apparently we have so many values in the underlying templates, so we always need to pass that `.Values` attribute. That was not the case when I did the first tests, so that's why I changed that. 

Since we need the `.Values` in a lot of cases, the original implementation is fine - I rolled it back.